src/main/java/life/qbic/portal/subscription/persistence/Subscriptions.java

```diff
@@ -28,16 +29,14 @@
 @Component
 class Subscriptions implements SubscriptionService {
 
-  public static final String CANCEL_ENDPOINT = "/subscription/cancel";
-  private final String serviceUrlBase;
+  public static final String CANCEL_ENDPOINT = "/subscriptions";
   private final String serviceEndPoint;
   Logger logger = LoggerFactory.getLogger(this.getClass());
   final ResponseHandler<String> responseHandler =
       response -> {
         int status = response.getStatusLine().getStatusCode();
         if (status >= 200 && status < 300) {
-          HttpEntity entity = response.getEntity();
-          return entity != null ? EntityUtils.toString(entity) : null;
+          return status+"";
```

Comment:
Converting an integer to a String by adding a String is a bit overhead. Better use the corresponding String method `String.valueOf(Integer)`

Reply:
I think this can be simplified more as the status code is not consumed later except for logging purposes. The if statement can be flipped and only the `else` case is kept. The response handler can be made to return a boolean instead of throwing an exception that is caught, logged and thrown again.

Reply:
You are right, updated it:)